merge-pr.sh

```diff
@@ -1,36 +1,69 @@
 #!/bin/bash
 
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Vars we need
 tmpMessageFile=".git-tmp-message.txt"
 tmpGemfile="Gemfile"
 tmpVendorDir="vendor"
 
+# Clean up
 rm -fr ${tmpMessageFile} ${tmpGemfile} ${tmpGemfile}.lock .bundle
 
+# Stop executing when we encounter errors
 set -e
 
+# Check if a pull request id was specified
 prId=$1
-
 if [ -z ${prId} ]; then
   echo "No PR number specified. Quiting."
   exit 1
 fi
 
+# Get ghi
 echo "source 'https://rubygems.org'" >> ${tmpGemfile}
 echo "gem 'ghi'" >> ${tmpGemfile}
-bundle install --path=${tmpVendorDir}/bundle --binstubs=${tmpVendorDir}/bin
+bundle install --path=${tmpVendorDir}/bundle --binstubs=${tmpVendorDir}/bin 2>&1 >/dev/null
 
+# Get vars from the GigHub API
 prIssue=$(bundle exec ghi show ${prId} | head -n 2)
 prAuthor=$(echo $prIssue | sed -e "s/^[^\@]*//" -e "s/ .*$//")
 prTitle=$(echo $prIssue | sed -e "s/^[^[:alpha:]]*: //" -e "s/ @.*$//")
 
-echo "Merge pull request #${prId} from ${prAuthor}" >> ${tmpMessageFile}
+# Construct the subject line of the commit message
+prSubject=$(echo "Merging #${prId} (${prAuthor}) ${prTitle}" | awk '{print substr($0,1,59)}')
+
+# Add dots when we shortened the subject
+subjectSize=${#prSubject}
+if [ ${subjectSize} -eq 59 ]; then
+    prSubject+=".."
+fi
+
```

Comment:
Example:

commit b6b60474948e40cc828f425a160a064e1735f383
Merge: ab4d9fc 5f87e9c
Author: Remi Bergsma <github@remi.nl>
Date:   Tue Aug 4 15:51:37 2015 +0200

    Merging #651 (@borisroman) CLOUDSTACK-8703: Fixed issue whe..
    
    * pr/651:
      CLOUDSTACK-8703: Fixed issue when listing directory on S3, it would only return objectSummaries when the anwser from the S3 System was truncated.
    
    Signed-off-by: Remi Bergsma <github@remi.nl>